### DISCUSSION TITLE
TypeError: options is undefined

### DISCUSSION THREAD

**Discussion by xinzhang20241216** [eclipse-theia/theia#14675] [General] [answered]
I follow the link
https://theia-ide.org/docs/frontend_application_contribution/

to write the below code, I got the error "TypeError: options is undefined", where is wrong?

```
@injectable()
export class BhsContribution extends AbstractViewContribution<BhsWidget> 
implements FrontendApplicationContribution{

    @inject(FrontendApplicationStateService)
    protected readonly stateService: FrontendApplicationStateService;
    
    async initializeLayout(app: FrontendApplication): Promise<void> {
        await this.openView();
    }

}
```

![image](https://github.com/user-attachments/assets/37f6a151-0b5e-40f4-b269-6385e054c4ff)
![image](https://github.com/user-attachments/assets/0cf83183-8c63-40f2-a54d-c056a66cdb85)

**Comment by msujew** [comment]
Hey @xinzhang20241216,

You need to supply the super class with an options argument in the constructor. See [this guide](https://theia-ide.org/docs/widgets/).

**Reply by xinzhang20241216** [reply]
Hi @msujew, thanks for your hint, now the exception disappear after adding one constructor as below screencut, but if my understanding right, the below code means when I first time open http://localhost:3000, the page will be show up, the question is why it does not show up? I must select the menu view-> my widget, the widget(view) can be open, so what else do I need to do if I want to see my widget when I first time open the link http://localhost:3000

```
    async initializeLayout(app: FrontendApplication): Promise<void> {
        await this.openView();
    }
```

![image](https://github.com/user-attachments/assets/88b39b09-306f-42e2-9ee3-11d2cdc17188)

![image](https://github.com/user-attachments/assets/b04125ae-9b1e-4e7e-8140-92c4a0d86fb6)

![image](https://github.com/user-attachments/assets/8780f6f9-b4c0-497c-9bc2-1328619a54cd)

**Reply by msujew** [reply, marked as answer]
You could run the `BhsCommand` command to open it when the application startsup. You could create a `FrontendApplicationContribution` for that.

Alternatively I would recommend to extract such functionality into a modal, see the [AbstractDialog](https://eclipse-theia.github.io/theia/docs/next/classes/core.AbstractDialog-1.html) api.

**Reply by xinzhang20241216** [reply]
Great thanks, as you mentioned, I used the below code, the widget was opened when open the http://localhost:3000 in the first time.

```
    async onStart(app: FrontendApplication): Promise<void> {
        this.stateService.reachedState('ready').then(
            () => this.openView({ reveal: true })
        );
    }
```
